assembl/static2/js/app/pages/idea.jsx

```diff
@@ -16,6 +16,50 @@ import ColumnsView from '../components/debate/multiColumns/columnsView';
 import ThreadView from '../components/debate/thread/threadView';
 import { DeletedPublicationStates } from '../constants';
 
+const deletedPublicationStates = Object.keys(DeletedPublicationStates);
+
+const sortByCreationDateDesc = (a, b) => {
+  if (a.creationDate > b.creationDate) {
+    return -1;
+  }
+  if (a.creationDate === b.creationDate) {
+    return 0;
+  }
+  return 1;
+};
+
+/*
+ * From a post, get the latest creationDate of live descendants and self
+ */
+const getLatest = (post) => {
+  let maxDate = post.creationDate;
+  if (post.children.length === 0) {
+    if (deletedPublicationStates.indexOf(post.publicationState) > -1) {
+      return null;
+    }
+    return maxDate;
+  }
+  post.children.forEach((p) => {
+    const date = getLatest(p);
+    if (date && date > maxDate) {
+      maxDate = date;
+    }
+  });
+  return maxDate;
+};
+
+const sortByCreationDateDescOfLastDescendant = (a, b) => {
```

Comment:
Likewise, should rename to `creationDateLastDescendantComparator` or something

Reply:
This is indeed a better name.